frontend/src/services/backend/club_backend.ts

```diff
@@ -0,0 +1,66 @@
+import { ClubInterface } from "../backend_service_interface/backend_service_interface"
+
+/**
+ * Directly connects with the servlet backend to perform club operations.
+ */
+export class ClubBackendService {
+  /** Creates a club on the backend given the club object. */
+  async createClub(club: ClubInterface): Promise<ClubInterface> {
+    const response = await fetch("/api/create-club", {
+      method: "POST",
+      body: JSON.stringify(club),
+    });
+    return response.json();
+  }
+
+  /** Retrieves a list of clubs with some relation to a user. */
+  async listClubs(
+      token: string,
+      membership: "member" | "not member"): Promise<ClubInterface[]> {
+    const listClubsUrl = "/api/list-clubs?" + new URLSearchParams({
+      idToken: token,
+      "membershipStatus": membership,
+    }).toString();
+    const response = await fetch(listClubsUrl, {
+      method: "GET",
+    });
+    return response.json();
+  }
+
+  /** Removes a user from a club. */
+  async leaveClub(clubId: string, token: string): Promise<number> {
```

Comment:
nit: Update the JSDoc comment to say what this number is. Same below

Reply:
Very good point! Fix is pushed for both function :).